### PR TITLE
feat: break down AstroErrorData in multiple objects

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

This PR removes the `AstroDataError` object and instead defines each error as an exported object.

Few things I had to change to adapt some logic:
- since we no longer have a top-level object, we cannot extract the name of the error at runtime. So I created a new `"name"` property, which is essentially the name of the error itself;
- `name` is now mandatory, so some `create*error` factory functions add that `"name"`. It's a bit redundant;
- the code that catches the error at runtime and shows the Astro overlay still needs to understand the name of the error, so I used a trick like `import * as AstroErrorData`. I tested locally with a project, and it still works;
- I turned off, for now, the action that checks the documentation

## Testing

- I updated some test case that was failing. The rest of the tests should pass.
- I manually used a project and intentionally created some code that fired an Astro error and checked that the overlays still showed all the relevant information.
- I manually built an edge middleware test case and checked that the bundle contained only the errors imported by the middleware. It only contains four objects, three shaking works :)

## Docs

I didn't add a changelog because this is an internal refactor.

### CODE REVIEW

**Review by bluwy — COMMENTED** [review]
One comment about the exports, but the general change LGTM!

File: packages/astro/src/core/errors/index.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
 export type { ErrorLocation, ErrorWithMetadata } from './errors';
-export { AstroErrorData } from './errors-data.js';
```
[the remark this answers is not in the document]
Comment (ematipico) [reply packages/astro/src/core/errors/index.ts:2]: Unfortunately, it's not possible. I already tried this approach, and it doesn't work for TypeScript.

When import `AstroErrorData` **from `index.js`**, TypeScript loses track of the objects.

```ts
import { AstroErrorData } from "../core/errors/index.js";

AstroErrorData.UnknownError
//						  ^^^^^^^^^^ <= compiler error, can't find UnknownError 
```

That's why we use `import *` directly from `errors-data.js`. Erika agreed with this workaround and flagged it as a valid exception.

[the remark this answers is not in the document]
Comment (bluwy) [reply packages/astro/src/core/errors/index.ts:2]: I don't see that error locally. Is it coming from your IDE or from running `tsc` when typechecking?

[the remark this answers is not in the document]
Comment (ematipico) [reply packages/astro/src/core/errors/index.ts:2]: Probably IDE error

**Review by bluwy — APPROVED** [review]
Looks great! Thanks for making the change.

EDIT: Looks like there's one test fail.

**Pull request merged**